pyhap/accessory.py

```diff
@@ -273,6 +290,43 @@ def remove_paired_client(self, client_uuid):
     def paired(self):
         return len(self.paired_clients) > 0
 
+    @property
+    def xhm_uri(self):
+        """Generates the X-HM:// uri (Setup Code URI)
+        @rtype: str
+        """
+        buffer = bytearray(b'\x00\x00\x00\x00\x00\x00\x00\x00')
+
+        value_low = int(self.pincode.replace(b'-', b''), 10)
+        value_low |= 1 << 28
+        struct.pack_into('>L', buffer, 4, value_low)
+
+        if self.category & 1:
```

Comment:
Could you expand this to `if self.category == Category.OTHER`?

Reply:
Done in next commit.